vpr/src/draw/search_bar.cpp

```diff
@@ -257,45 +300,83 @@ void highlight_blocks(ClusterBlockId clb_index) {
+    int max_depth = draw_state->show_blk_internal;
+    if (atom_block_depth > max_depth) return false;
 
-    for (tptr = route_ctx.trace[net_id].head; tptr != nullptr; tptr = tptr->next) {
-        draw_state->net_color[net_id] = ezgl::MAGENTA;
-    }
+    //Highlighting block
+    get_selected_sub_block_info().set(atom_block_pb, cl_blk);
+    app->refresh_drawing();
+    return true;
 }
 
-void highlight_nets(std::string net_name) {
-    auto& cluster_ctx = g_vpr_ctx.clustering();
-
-    ClusterNetId net_id = ClusterNetId::INVALID();
-    net_id = cluster_ctx.clb_nlist.find_net(net_name);
-
-    if (net_id == ClusterNetId::INVALID()) {
-        warning_dialog_box("Invalid Net Name");
-        return; //name not exist
+/**
+ * @brief Recursively looks through pb graph to find block w. given name
+ * 
+ * @param name name of block being searched for
+ * @param pb current node to be examined
+ * @return t_pb* t_pb ptr of block w. name "name". Returns nullptr if nothing found
+ */
+t_pb* find_atom_block_in_pb(std::string name, t_pb* pb) {
+    //Checking if block is one being searched for
+    std::string pbName(pb->name);
+    if (pbName == name) return pb;
+    //If block has no children, returning
+    if (pb->child_pbs == nullptr) return nullptr;
+
+    int num_child_types = pb->get_num_child_types();
+    //Iterating through all child types
+    for (int i = 0; i < num_child_types; ++i) {
+        if (pb->child_pbs[i] == nullptr) continue;
+        int num_children_of_type = pb->get_num_children_of_type(i);
+        //Iterating through all of pb's children of given type
+        for (int j = 0; j < num_children_of_type; ++j) {
+            t_pb* child_pb = &pb->child_pbs[i][j];
+            //If child exists, recursively calling function on it
+            if (child_pb->name != nullptr) {
+                t_pb* subtree_result = find_atom_block_in_pb(name, child_pb);
+                //If a result is found, returning it to top of recursive calls
+                if (subtree_result != nullptr) {
+                    return subtree_result;
+                }
+            }
+        }
     }
-
-    highlight_nets(net_id); //found net
+    return nullptr;
 }
 
-void highlight_blocks(std::string block_name) {
-    auto& cluster_ctx = g_vpr_ctx.clustering();
+void highlight_nets(ClusterNetId net_id) {
+    t_trace* tptr;
+    auto& route_ctx = g_vpr_ctx.routing();
 
-    ClusterBlockId block_id = ClusterBlockId::INVALID();
-    block_id = cluster_ctx.clb_nlist.find_block(block_name);
+    t_draw_state* draw_state = get_draw_state_vars();
 
-    if (block_id == ClusterBlockId::INVALID()) {
-        warning_dialog_box("Invalid Block Name");
-        return; //name not exist
-    }
+    //If routing does not exist return
+    if (int(route_ctx.trace.size()) == 0) return;
 
-    highlight_blocks(block_id); //found block
+    for (tptr = route_ctx.trace[net_id].head; tptr != nullptr; tptr = tptr->next) {
```

Comment:
This loop is redundant; it sets the draw_state->net_color[net_id] many times (once per routing resource in its traceback) but it can just do it once.